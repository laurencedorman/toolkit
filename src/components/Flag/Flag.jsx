import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import France from './Flags/France'
import styles from './Flag.module.scss';

const Flag = ({
  onClick,
  onKeyPress,
  className,
  size,
  fill,
  stroke,
  dataQa,
  component: Component,
}) => {
  const classNames = cn(
    styles.container,
    className,
  );

  return (
    <div
      className={classNames}
      onClick={onClick}
      onKeyPress={onKeyPress}
      role={onClick ? 'button' : 'img'}
      data-qa={dataQa}
    >
      <svg
        className={styles.svg}
        width={`${String(size)}px`}
        height={`${String(size)}px`}
        viewBox="0 0 32 32"
        role="img"
        fill={fill}
        xmlns="http://www.w3.org/2000/svg"
      >
        <Component />
      </svg>
    </div>
  );
};


Flag.propTypes = {
  onClick: PropTypes.func,
  onKeyPress: PropTypes.func,
  className: PropTypes.string,
  size: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  fill: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
  ]),
  stroke: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
  ]),
  dataQa: PropTypes.string,
  component: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]),
};

Flag.defaultProps = {
  onClick: null,
  onKeyPress: null,
  className: '',
  size: 32,
  fill: '#0c193a',
  stroke: '#0c193a',
  dataQa: '',
};

export default Flag;
