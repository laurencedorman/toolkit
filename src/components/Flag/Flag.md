Flag

  * Format SVG

```js
import Belgium from './Flags/Belgium';
import France from './Flags/France';
import Germany from './Flags/Germany';
import Italy from './Flags/Italy';
import Netherlands from './Flags/Netherlands';
import Spain from './Flags/Spain';
import Switzerland from './Flags/Switzerland';
import UnitedKingdom from './Flags/UnitedKingdom';

<div style={{ display: 'flex', width: 400, justifyContent: 'space-around' }}>
  <Flag component={Belgium} />
  <Flag component={France} />
  <Flag component={Germany} />
  <Flag component={Italy} />
  <Flag component={Netherlands} />
  <Flag component={Spain} />
  <Flag component={Switzerland} />
  <Flag component={UnitedKingdom} />
</div>
```
