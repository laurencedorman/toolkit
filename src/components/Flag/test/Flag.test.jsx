import React from 'react';
import { shallow } from 'enzyme';
import Flag from '../Flag';
import France from '../Flags/France';

describe('Flag', () => {
  let wrapper;

  wrapper = shallow(<Flag name="document" />);

  it('should render without crash with layoutContainer', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should render different role', () => {
    const click = jest.fn();
    wrapper.setProps({ onClick: click });
  });

  it('should render different svg with special name', () => {
    wrapper = shallow(<Flag component={France} />);
  });
});
