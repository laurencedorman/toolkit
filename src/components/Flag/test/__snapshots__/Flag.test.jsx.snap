// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag template should contain role button when onclick is undefined 1`] = `"<div class=\\"container\\" role=\\"button\\" data-qa=\\"data-qa\\"><svg class=\\"svg\\" width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 32 32\\" role=\\"img\\" fill=\\"#0c193a\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"#FFDA44\\" d=\\"M21.6 1C19.8.4 18 0 16 0s-3.8.4-5.6 1L9 16l1.4 15c1.7.6 3.6 1 5.6 1s3.8-.4 5.6-1L23 16 21.6 1z\\"></path><path fill=\\"#D80027\\" d=\\"M32 16c0-6.9-4.3-12.7-10.4-15v30C27.7 28.7 32 22.9 32 16z\\"></path><path d=\\"M0 16c0 6.9 4.3 12.7 10.4 15V1C4.3 3.3 0 9.1 0 16z\\"></path></svg></div>"`;

exports[`Flag template should render different svg with special name 1`] = `"<div class=\\"container\\" role=\\"img\\" data-qa=\\"data-qa\\"><svg class=\\"svg\\" width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 32 32\\" role=\\"img\\" fill=\\"#0c193a\\" xmlns=\\"http://www.w3.org/2000/svg\\"><circle fill=\\"#F0F0F0\\" cx=\\"16\\" cy=\\"16\\" r=\\"16\\"></circle><path fill=\\"#D80027\\" d=\\"M32,16c0-6.9-4.3-12.7-10.4-15v30C27.7,28.7,32,22.9,32,16z\\"></path><path fill=\\"#0052B4\\" d=\\"M0,16c0,6.9,4.3,12.7,10.4,15V1C4.3,3.3,0,9.1,0,16z\\"></path></svg></div>"`;

exports[`Flag template should render without crash with layoutContainer 1`] = `"<div class=\\"container\\" role=\\"img\\" data-qa=\\"data-qa\\"><svg class=\\"svg\\" width=\\"32px\\" height=\\"32px\\" viewBox=\\"0 0 32 32\\" role=\\"img\\" fill=\\"#0c193a\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path fill=\\"#FFDA44\\" d=\\"M21.6 1C19.8.4 18 0 16 0s-3.8.4-5.6 1L9 16l1.4 15c1.7.6 3.6 1 5.6 1s3.8-.4 5.6-1L23 16 21.6 1z\\"></path><path fill=\\"#D80027\\" d=\\"M32 16c0-6.9-4.3-12.7-10.4-15v30C27.7 28.7 32 22.9 32 16z\\"></path><path d=\\"M0 16c0 6.9 4.3 12.7 10.4 15V1C4.3 3.3 0 9.1 0 16z\\"></path></svg></div>"`;
